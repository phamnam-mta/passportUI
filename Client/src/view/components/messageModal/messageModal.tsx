import * as React from "react";
import { Text, PrimaryButton, DefaultButton } from "@fluentui/react";

import Modal from "view/components/modal/modal";

interface IMessageModalProps {
    isOpen: boolean;
    title: string;
    body: React.ReactNode;
    actionButtonDisabled?: boolean;
    actionButtonText?: string;
    rejectButtonText?: string;
    isLoading?: boolean;
    loadingText?: string;
    width?: number;
    onActionButtonClick?: () => void;
    onClose?: () => void;
    onDismiss?: () => void;
}

export class MessageModal extends React.PureComponent<IMessageModalProps> {
    public render() {
        const {
            isOpen,
            title,
            body,
            width,
            isLoading,
            actionButtonDisabled,
            actionButtonText,
            rejectButtonText,
            loadingText,
            onActionButtonClick,
            onClose,
            onDismiss,
        } = this.props;
        return (
            <Modal
                isOpen={isOpen}
                isLoading={isLoading}
                header={<Text variant="xLarge">{title}</Text>}
                body={body}
                footer={<div></div>}
                buttonGroup={
                    <>
                        {onActionButtonClick && (
                            <PrimaryButton
                                text={actionButtonText}
                                disabled={actionButtonDisabled}
                                onClick={onActionButtonClick}
                            />
                        )}
                        <DefaultButton text={rejectButtonText ? rejectButtonText : "Cancel"} onClick={onClose} />
                    </>
                }
                loadingText={loadingText}
                width={width}
                onClose={onClose}
                onDismiss={onDismiss}
            />
        );
    }
}

export default MessageModal;
