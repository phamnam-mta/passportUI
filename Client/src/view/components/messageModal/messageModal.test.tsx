import * as React from "react";
import { shallow } from "enzyme";
import { MessageModal } from "./messageModal";

describe("<MessageModal />", () => {
    let baseProps;

    beforeEach(() => {
        baseProps = {
            isOpen: true,
            title: "<title>",
            body: <div>Hello World</div>,
            width: 700,
            actionButtonDisabled: false,
            actionButtonText: "Done",
            onActionButtonClick: jest.fn(),
            onClose: jest.fn(),
        };
    });

    describe("Rendering", () => {
        it("should render correctly", () => {
            const wrapper = shallow(<MessageModal {...baseProps} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when there's no action button", () => {
            const props = { ...baseProps, onActionButtonClick: undefined };

            const wrapper = shallow(<MessageModal {...props} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when it is loading", () => {
            const props = { ...baseProps, isLoading: true };

            const wrapper = shallow(<MessageModal {...props} />);

            expect(wrapper).toMatchSnapshot();
        });

        it("should match snapshot, when there is rejectButtonText", () => {
            const props = { ...baseProps, rejectButtonText: "No" };

            const wrapper = shallow(<MessageModal {...props} />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Event Handling", () => {
        it("should trigger onActionButtonClick, when action button is clicked", () => {
            const wrapper = shallow(<MessageModal {...baseProps} />);

            const ButtonGroup: any = () => wrapper.find("Modal").prop("buttonGroup");
            const button = shallow(<ButtonGroup />).find("CustomizedPrimaryButton");
            button.simulate("click");

            expect(baseProps.onActionButtonClick).toBeCalledTimes(1);
        });

        it("should trigger onClose, when close button is clicked", () => {
            const wrapper = shallow(<MessageModal {...baseProps} />);

            const ButtonGroup: any = () => wrapper.find("Modal").prop("buttonGroup");
            const button = shallow(<ButtonGroup />).find("CustomizedDefaultButton");
            button.simulate("click");

            expect(baseProps.onClose).toBeCalledTimes(1);
        });
    });
});
