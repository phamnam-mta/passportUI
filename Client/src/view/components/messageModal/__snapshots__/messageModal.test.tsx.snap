// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageModal /> Rendering should match snapshot, when it is loading 1`] = `
<Modal
  body={
    <div>
      Hello World
    </div>
  }
  buttonGroup={
    <React.Fragment>
      <CustomizedPrimaryButton
        disabled={false}
        onClick={[MockFunction]}
        text="Done"
      />
      <CustomizedDefaultButton
        onClick={[MockFunction]}
        text="Cancel"
      />
    </React.Fragment>
  }
  footer={<div />}
  header={
    <Text
      variant="xLarge"
    >
      &lt;title&gt;
    </Text>
  }
  isLoading={true}
  isOpen={true}
  onClose={[MockFunction]}
  width={700}
/>
`;

exports[`<MessageModal /> Rendering should match snapshot, when there is rejectButtonText 1`] = `
<Modal
  body={
    <div>
      Hello World
    </div>
  }
  buttonGroup={
    <React.Fragment>
      <CustomizedPrimaryButton
        disabled={false}
        onClick={[MockFunction]}
        text="Done"
      />
      <CustomizedDefaultButton
        onClick={[MockFunction]}
        text="No"
      />
    </React.Fragment>
  }
  footer={<div />}
  header={
    <Text
      variant="xLarge"
    >
      &lt;title&gt;
    </Text>
  }
  isOpen={true}
  onClose={[MockFunction]}
  width={700}
/>
`;

exports[`<MessageModal /> Rendering should match snapshot, when there's no action button 1`] = `
<Modal
  body={
    <div>
      Hello World
    </div>
  }
  buttonGroup={
    <React.Fragment>
      <CustomizedDefaultButton
        onClick={[MockFunction]}
        text="Cancel"
      />
    </React.Fragment>
  }
  footer={<div />}
  header={
    <Text
      variant="xLarge"
    >
      &lt;title&gt;
    </Text>
  }
  isOpen={true}
  onClose={[MockFunction]}
  width={700}
/>
`;

exports[`<MessageModal /> Rendering should render correctly 1`] = `
<Modal
  body={
    <div>
      Hello World
    </div>
  }
  buttonGroup={
    <React.Fragment>
      <CustomizedPrimaryButton
        disabled={false}
        onClick={[MockFunction]}
        text="Done"
      />
      <CustomizedDefaultButton
        onClick={[MockFunction]}
        text="Cancel"
      />
    </React.Fragment>
  }
  footer={<div />}
  header={
    <Text
      variant="xLarge"
    >
      &lt;title&gt;
    </Text>
  }
  isOpen={true}
  onClose={[MockFunction]}
  width={700}
/>
`;
